Makefile

```diff
@@ -31,13 +31,22 @@ verify: patch fmt vet
 unit:
 	@echo "TODO UNIT TEST"
 
-go-deploy-manifests: manifests$(SUFFIX)
-	go run test/deploy/deploy.go -path ./manifests$(SUFFIX)
-
-go-undeploy-manifests:
-	go run test/undeploy/undeploy.go -path ./manifests$(SUFFIX)
-
-test-e2e-upgrade: go-deploy-manifests
+CRDS := $(shell echo manifests/[0-9][0-9][0-9][0-9]_customresourcedefinition_*.yaml)
+
+deploy-manifests: manifests$(SUFFIX)
+	# First, create CRDs
+	for c in $(CRDS); do kubectl apply -f $${c}; done
+	# Wait for CRDs to be established in the API server, otherwise adding the corresponding CRs could fail
+	for c in $(CRDS); do kubectl wait --for condition=established --timeout=60s -f $${c}; done
+	# Apply all the rest
+	kubectl apply --wait -f ./manifests$(SUFFIX)
+
+undeploy-manifests: manifests$(SUFFIX)
+	kubectl delete --wait --ignore-not-found -f ./manifests$(SUFFIX)
+
+# TODO: obsolete targets; remove
+go-deploy-manifests: deploy-manifests
+go-undeploy-manifests: undeploy-manifests
```

Comment:
I would do this in a separate PR, this is not strictly related to simplifying and fixing the CI